src/core/dbus-execute.c

```diff
@@ -3563,6 +3594,61 @@ int bus_exec_context_set_transient_property(
 
                 return 1;
 
+        } else if (STR_IN_SET(name, "StateDirectoryQuota", "CacheDirectoryQuota", "LogsDirectoryQuota")) {
+                uint64_t quota_absolute = UINT64_MAX;
+                uint32_t quota_scale = UINT32_MAX;
+                bool quota_enforce;
+
+                r = sd_bus_message_enter_container(message, 'r', "tub");
+                if (r < 0)
+                        return r;
+
+                r = sd_bus_message_read(message, "t", &quota_absolute);
+                if (r < 0)
+                        return r;
+
+                r = sd_bus_message_read(message, "u", &quota_scale);
+                if (r < 0)
+                        return r;
+
+                r = sd_bus_message_read(message, "b", &quota_enforce);
+                if (r < 0)
+                        return r;
+
+                r = sd_bus_message_exit_container(message);
+                if (r < 0)
+                        return r;
```

Comment:
replace these five calls by:

```c
r = sd_bus_message_read(message, "(tub)", &qota_absolute, &quota_scale, &quota_enforce));
```

i.e. no need to read these piecemeal, you can read the whole struct at once